### PR TITLE
Refactored to work with zend-mvc >= 2.7

### PR DESCRIPTION
**Pull request by kargpfen**
Change config to work with new zend-mvc/src/Router/RouteInvokableFactory.php

### CODE REVIEW

**Comment by mtudor** [comment]
::class adds a dependency on PHP 5.5+ - not necessarily a show stopper but we should consider it.  I'm not up to speed on my ZF3 changes yet - can you give us a quick run down on the benefits of the RouteInvokableFactory (or point us to somewhere that does) and also whether this change will be backwards compatible with ZF2 - I'm guessing that it might not?

**Comment by kargpfen** [comment]
PHP 5.5 is more than 3 years old and since ZF 2.5 „::class“ is used at many source code passages. But of course I can roll this back.

The actual problem: since zend-mvc 2.7 the RoutePluginManager was changed in a way that invokable Route-Classes are not instanced automatically via factory method (RouterInterface) but via AbstractPluginManager from zend-servicemanager. Before, the RoutePluginManager had an own „createFromInvokable“-method where $invokable::factory($this->creationOptions)  was called.

Since zend-mvc 2.7 invokables are generated by AbstractPluginManager::createFromInvokable and here „new $invokable($options)“ is used which ends in the problem that SkippableSegment::_construct is instanced with only one array parameter and errors are thrown. But RouteInvokableFactory generatest the class as usual via the „factory“ method.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,18 @@
 TccSkippableSegment
 =======
 
-This module is based on the work by: @towr (https://gist.github.com/towr/3320fb4ee13dfd079890).
+This ZF3 module is based on the work by: @towr (https://gist.github.com/towr/3320fb4ee13dfd079890).
```
Comment (shipleyr) [inline README.md:4]: This is more than a ZF3 module. It should be compatible with ZF2 as well.

File: autoload_classmap.php [hunk 2]
```diff
@@ -1,6 +0,0 @@
-<?php
```
Comment (shipleyr) [inline autoload_classmap.php:1]: Why has this file been removed?

File: composer.json [hunk 3]
```diff
@@ -14,15 +14,11 @@
         }
     ],
     "require"     : {
-        "php"                         : ">=5.3.0",
-        "zendframework/zendframework" : "~2.3"
+        "zendframework/zend-mvc": "^3.0.2"
```
Comment (shipleyr) [inline composer.json:17]: Too high. This needs to work with earlier versions as well.

File: Module.php [hunk 4]
```diff
@@ -1,29 +0,0 @@
...
-
-    public function getAutoloaderConfig()
-    {
-        return array(
-            'Zend\Loader\ClassMapAutoloader' => array(
-                __DIR__ . '/autoload_classmap.php',
-            ),
-        );
-    }
-}
```
Comment (shipleyr) [inline Module.php:29]: Why have you deleted this?

File: src/Module.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                'aliases' => [
+                    'SkippableSegment' => SkippableSegment::class,
+                ],
+                'factories' => [
+                    SkippableSegment::class => RouteInvokableFactory::class,
+                ],
+            ],
+        ];
+    }
+}
```
Comment (shipleyr) [inline src/Module.php:33]: Please move back to previous location.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+/vendor/
+/.idea/
+/composer.lock
```
Comment (shipleyr) [inline .gitignore:3]: Please remove .gitignore entirely.

**Comment by shipleyr** [comment]
Sorry to say there is a lot that needs updating with this PR. Please can you address these and create another PR with the changes included.

**Comment by MatthiasKuehneEllerhold** [comment]
Almost (all?) all things you commented on were introduced because of my PR for ZF3 compatibility. So I'd like to give my reasoning behind these changes.

What I did was a refactoring to make this module a ZF3 module. So that new version(s) of TccSkippableSegment (after 1.0.1) will be used for ZF3 and 1.0.1 will be used for ZF2. 
Having it support both versions would introduces a heavy compatibiliy layer thats just not worth it (in my opinion). 

The downside would be that you'd have to support both branches (a ZF2- and a ZF3-branch). But realisticly you've had 2 releases in the last 16 months and I dont think this will change much.

Ive removed the zend-autoloader because composer is the way to go nowadays and provides the exact same functionality (but better). It might have been a bit overzealous doing that. ;)

**Pull request closed without merging**